app/main.py

```diff
@@ -1,53 +1,92 @@
-from typing import Tuple
+import random
+from enum import Enum
+from typing import Dict, List, Tuple
 
 from fastapi import FastAPI, HTTPException, status
 from fastapi.middleware.cors import CORSMiddleware
 from pydantic import BaseModel
-import random
-from enum import Enum
 
 app = FastAPI()
 
 app.add_middleware(
     CORSMiddleware,
     allow_origins=[
-        "http://localhost:5173", # frontend dev
-        "http://localhost:8080", # frontend Docker
+        "http://localhost:5173",  # frontend dev
+        "http://localhost:8080",  # frontend Docker
     ],
     allow_credentials=True,
     allow_methods=["*"],
     allow_headers=["*"],
 )
 
-DRONE_QUANTITY = 5
-DRONE_CENTER = [54.39, -0.937]
-DRONE_STATUSES = ["idle", "flying", "unknown"]
-
-@app.get("/")
-async def hello_world():
-    return {"response": "Hello World"}
+DroneId = str
+LatLon = Tuple[float, float]
 
 
 class DroneData(BaseModel):
-    id: str
     status: str
     battery: int
     lastUpdate: str
-    lastSeen: Tuple[float, float]
+    lastSeen: LatLon
+
 
+DRONE_STATUSES = ["idle", "flying", "unknown"]
+DRONES: Dict[DroneId, DroneData] = dict()
 
 
-@app.get("/get_drone_status")
-async def get_drone_status():
-    def create_random_drone(id: str):
-        data = DroneData(
-            id=id,
-            status=str(random.choice(DRONE_STATUSES)),
-            battery=random.randint(0, 100),
-            lastUpdate="",
-            lastSeen=(DRONE_CENTER[0] + random.random() * 0.05, DRONE_CENTER[1] + random.random() * 0.05)
+@app.get("/")
+async def hello_world():
+    return {"response": "Hello World"}
+
+
+# get the status (position) of all the drones
+# called by the frontend
+@app.get("/drone_status")
+async def get_drone_status() -> List[Tuple[DroneId, DroneData]]:
+    DRONE_QUANTITY = 5
+    DRONE_CENTER = [54.39, -0.937]
+
+    # TODO: send from internal storage
+    def create_random_drone(id: DroneId) -> Tuple[DroneId, DroneData]:
+        return (
+            id,
+            DroneData(
+                status=str(random.choice(DRONE_STATUSES)),
+                battery=random.randint(0, 100),
+                lastUpdate="",
+                lastSeen=(
+                    DRONE_CENTER[0] + random.random() * 0.05,
+                    DRONE_CENTER[1] + random.random() * 0.05,
+                ),
+            ),
         )
-        return data
-    drones = [create_random_drone(str(i+1)) for i in range(DRONE_QUANTITY)]
 
-    return drones
+    return [create_random_drone(str(i + 1)) for i in range(DRONE_QUANTITY)]
+
+
+class TargetCircle(BaseModel):
```

Comment:
Nit: I would move this up to the top with the other classes

Reply:
Agreed that it's messy. I've got commits coming up which move this into another file with the other search methods to keep the logic away from the endpoints (which are essentially just function call wrappers). I'm a little reluctant to do that refactor here since it might muck my upstream stuff up; rest assured that in the "real" code it's no longer in that position.

Reply:
Ah nice, lgtm then